### PR TITLE
Add display-modes extension

### PR DESCRIPTION
**Pull request by codifilo**
## Description

A extension that lists your plugged displays and for each one it lists the available modes (resolution, scale and refresh rate) and allows to switch to a new display mode.

## Screencast

https://github.com/raycast/extensions/assets/2012509/c94b8b50-74bd-47bd-b920-95aefefdb0e7

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/display-modes/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import { ActionPanel, List, Action, useNavigation, showHUD, KeyEquivalent } from "@raycast/api";
```
Comment (pernielsentikaer) [inline extensions/display-modes/src/index.tsx:1]:
```suggestion
import { ActionPanel, List, Action, useNavigation, showHUD, Icon } from "@raycast/api";
```

File: extensions/display-modes/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+          return (
+            <List.Item
+              key={index}
+              icon={modeInfo.isEqual ? "selected-mode-icon.png" : "mode-icon.png"}
+              title={formatDisplayMode(modeInfo.mode)}
+              subtitle={modeInfo.isEqual ? "Current Mode" : undefined}
+              actions={
+                <ActionPanel>
+                  <Action
+                    title="Change Display Mode"
```
Comment (pernielsentikaer) [inline extensions/display-modes/src/index.tsx:68]:
```suggestion
                    title="Change Display Mode"
                    icon={Icon.Monitor}
```
Adding an icon makes it even more beautiful 🙂

File: extensions/display-modes/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        return (
+          <List.Item
+            key={index}
+            icon="display-icon.png"
+            title={formatTitle(display)}
+            subtitle={formatSubtitle(display)}
+            actions={
+              <ActionPanel>
+                <Action.Push
+                  title="Show Display Modes"
```
Comment (pernielsentikaer) [inline extensions/display-modes/src/index.tsx:104]:
```suggestion
                  title="Show Display Modes"
                  icon={Icon.Monitor}
```
Adding an icon makes it even more beautiful 🙂

File: extensions/display-modes/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                  target={detail(display)}
+                  {...(key ? { modifiers: ["opt"], key: key } : {})}
```
Comment (pernielsentikaer) [inline extensions/display-modes/src/index.tsx:106]:
```suggestion
                  {...(key ? { shortcut: { modifiers: ["opt"], key: key } } : {})}
```
You missed `shortcut` here, it should look like the above suggestion

If you hold  `⌘` for 400 ms you'll get the same behavior out of the box, I would actually just remove the above and rely on  `⌘+num` instead which is how Raycast works in every list 🙂

File: extensions/display-modes/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            <List.Item
+              key={index}
+              icon={modeInfo.isEqual ? "selected-mode-icon.png" : "mode-icon.png"}
+              title={formatDisplayMode(modeInfo.mode)}
+              subtitle={modeInfo.isEqual ? "Current Mode" : undefined}
+              actions={
+                <ActionPanel>
+                  <Action
+                    title="Change Display Mode"
+                    {...(key ? { modifiers: ["opt"], key: key } : {})}
```
Comment (pernielsentikaer) [inline extensions/display-modes/src/index.tsx:69]:
```suggestion
                    {...(key ? { shortcut: { modifiers: ["opt"], key: key } } : {})}
```
You missed `shortcut` here, it should look like the above suggestion

If you hold  `⌘` for 400 ms you'll get the same behavior out of the box, I would actually just remove the above and rely on  `⌘+num` instead which is how Raycast works in every list 🙂

File: extensions/display-modes/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
+import { environment, getPreferenceValues, Icon, Image, Keyboard, List } from "@raycast/api";
+import { execa, ExecaError } from "execa";
+import { chmod } from "fs/promises";
+import { join } from "path";
+import { DisplayInfo, Mode } from "./types";
+
+const preferences: Preferences = getPreferenceValues();
+
```
Comment (pernielsentikaer) [inline extensions/display-modes/src/utils.ts:8]:
```suggestion
import { environment } from "@raycast/api";
import { execa } from "execa";
import { chmod } from "fs/promises";
import { join } from "path";
import { DisplayInfo, Mode } from "./types";

```
Just removed unused imports

File: extensions/display-modes/src/utils.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    case 7:
+      return "7";
+    case 8:
+      return "8";
+    case 9:
+      return "9";
+    default:
+      return undefined;
+  }
+}
```
Comment (pernielsentikaer) [inline extensions/display-modes/src/utils.ts:76]:
```suggestion
```
If relying on default Raycast behavior this is not necessary 🙂

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

#### Suggestions (For later)
- Add a menubar command too 🙂
- Allow favorite modes in the list (if possible)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/display-modes/src/index.tsx (hunk 1 above)
Comment (codifilo) [reply extensions/display-modes/src/index.tsx]: Done!

File: extensions/display-modes/src/index.tsx (hunk 2 above)
Comment (codifilo) [reply extensions/display-modes/src/index.tsx:67]: Done. Looks better!

File: extensions/display-modes/src/index.tsx (hunk 3 above)
Comment (codifilo) [reply extensions/display-modes/src/index.tsx]: Done. Looks better!

File: extensions/display-modes/src/index.tsx (hunk 4 above)
Comment (codifilo) [reply extensions/display-modes/src/index.tsx]: Oh, yeah this is totally unnecessary then. I removed it.
Very convenient default behaviour from Raycast 👏

File: extensions/display-modes/src/index.tsx (hunk 5 above)
Comment (codifilo) [reply extensions/display-modes/src/index.tsx]: Oh, yeah this is totally unnecessary then. I removed it.
Very convenient default behaviour from Raycast 👏

File: extensions/display-modes/src/utils.ts (hunk 6 above)
Comment (codifilo) [reply extensions/display-modes/src/utils.ts]: Done.

File: extensions/display-modes/src/utils.ts (hunk 7 above)
Comment (codifilo) [reply extensions/display-modes/src/utils.ts]: Removed.

**Comment by codifilo** [comment]
@pernielsentikaer Thank you very much for your feedback.
I added the suggested changes to the PR.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**